### PR TITLE
Initial PATTERN.md

### PR DESCRIPTION
**Pull request by edwarnicke**

### CODE REVIEW

File: PATTERN.md [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+   and then
+4. Build
+5. Test
+6. Debug
+7. Runtime
+
+
+### Go
+If you are working on Network Service Mesh cmds that do not require vppagent, you can usually simply
+use a [golang base docker image](https://hub.docker.com/_/golang) to get Go.  However, in the case of
```
Comment (haiodo) [inline PATTERN.md:192]: Spire/spiffie is not working with goland base image unfortunately. I've used ubuntu as test image in compare to ligato base image.

**Comment by denis-tingajkin** [comment]
Only one moment here... Do we need to add this PATTERN.md to each cmd repository or we should put this pattern to some common place?

File: PATTERN.md (hunk 1 above)
Comment (edwarnicke) [reply PATTERN.md:192]: What’s the issue?

**Comment by edwarnicke** [comment]
> Do we need to add this PATTERN.md for each cmd repository or we can put this pattern to some common place?

I don't think so.  I wrote it here just because it was close to the initial example.  As we get more cmds done, we should figure out a good place to put it :)

File: PATTERN.md (hunk 1 above)
Comment (edwarnicke) [reply PATTERN.md:192]: @haiodo Have you tried this:
```dockerfile
FROM golang:1.14.3-alpine3.11 as alpine
COPY --from=gcr.io/spiffe-io/spire-agent:0.10.0 /opt/spire/bin /bin
COPY --from=gcr.io/spiffe-io/spire-server:0.10.0 /opt/spire/bin /bin
```
?

The tarball binaries for download are based on glibc.  golang alpine images use musl.  The spiffe-io containers are built on alpine, and thus use musl, and on cursory testing appear to work :)

**Pull request merged**